src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -930,7 +930,7 @@ public function setAttribute($key, $value)
         // If an attribute is listed as a "date", we'll convert it from a DateTime
         // instance into a form proper for storage on the database tables using
         // the connection grammar's date format. We will auto set the values.
-        elseif ($value && $this->isDateAttribute($key)) {
+        elseif (isset($value) && $this->isDateAttribute($key)) {
```

Comment:
Maybe it is better to use `! is_null($value)` here instead of `isset`, it is cleaner and consistent with [HasAttributes.php#L949](https://github.com/laravel/framework/blob/af9d6f3df83e7f5f4763902985bc5d02d6dad662/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L949) and [HasAttributes.php#L960](https://github.com/laravel/framework/blob/af9d6f3df83e7f5f4763902985bc5d02d6dad662/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L960).

Reply:
```suggestion
        elseif (! is_null($value) && $this->isDateAttribute($key)) {
```

Reply:
Done, thanks.